product_packaging_unit_price_calculator/wizards/product_package_price.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductPackagePrice(models.TransientModel):
+    _name = "product.package.price.wizard"
+    _description = "Wizard to compute unit price from packaging price"
+
+    def _default_product_tmpl_id(self):
+        return self.env["product.template"].browse(self._context.get("product_tmpl_id"))
+
+    def _default_product_pricelist_item_id(self):
+        if self._context.get("active_model") != "product.pricelist.item":
+            return False
+        return self.env["product.pricelist.item"].browse(self._context.get("active_id"))
+
+    def _default_product_supplierinfo_id(self):
+        if self._context.get("active_model") != "product.supplierinfo":
+            return False
+        return self.env["product.supplierinfo"].browse(self._context.get("active_id"))
+
+    product_id = fields.Many2one("product.template", default=_default_product_tmpl_id)
```

Comment:
use lambda to not reference the method's object directly for every default

Reply:
Changed, also not sure what it improves ?

Reply:
if you reference the object you cannot override it w/out re-defining the field

Reply:
But if you use lambda don't you have to redefine the field to override the default as well ?

Reply:
no, the lambda delegates to the method itself as `default=lambda self: self._default_whatever()`. This is the rule of thumb for such cases.

Reply:
ha, keep the function not just replace the function with a lambda one